Fixes an issue where datapacks appear twice in the dash

acron0 opened the pull request:
Also noticed that extra user data was being copied into certain items

acron0 commented:
:shipit:

Commit:
Fixes an issue where datapacks appear twice in the dash

Commit:
Added test

Commit:
Use a better base image to avoid installing Java8

tcoupland commented on src/cljs/witan/ui/controllers/datastore.cljs:
```diff
@@ -37,10 +37,15 @@
-    (data/swap-app-state! :app/datastore update :ds/file-metadata #(assoc % (:kixi.datastore.metadatastore/id new-datapack-meta) new-datapack-meta))))
+  (let [new-id (:kixi.datastore.metadatastore/id md)
+        new-meta (assoc-in md
+                           [:kixi.datastore.metadatastore/provenance :kixi/user] (data/get-user))
```
how come this isn't already present?

acron0 replied:
Because we don't send the user as part of the provenance, it's added by the datastore. This add function is transforming to make it look as if it's come from datastore.

tcoupland commented on src/cljs/witan/ui/data.cljs:
```diff
@@ -55,6 +55,13 @@
+  {:kixi.user/name nil
+   :kixi.user/username nil
+   :kixi.user/id nil
+   :kixi.user/groups [nil]
```
Wldn't an empty vector make more sense?

acron0 replied:
No, this is used as starting data.

Commit:
Fix dash spinner

Commit:
Invert behaviour here; always use the remote copy.

At some point we might end up with two functions, or an argument,
which controls this with more nuance
